perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java

```diff
@@ -2139,10 +2146,51 @@ public AttributeDefinition createAttribute(PerunSession sess, AttributeDefinitio
 			}
 		}
 
+		if (calculateDependencies) {
+			synchronized (dependenciesMonitor) {
+				handleAttributeModuleDependencies(sess, attribute);
+			}
+		}
+
 		getPerunBl().getAuditer().log(sess, "{} created.", attribute);
 		return getAttributesManagerImpl().createAttribute(sess, attribute);
 	}
 
+	/**
+	 * For given attribute finds its dependencies and adds them to the all maps containing any dependencies.
+	 *
+	 * @param sess session
+	 * @param attribute attribute
+	 * @throws InternalErrorException internal error
+	 */
+	private void handleAttributeModuleDependencies(PerunSession sess, AttributeDefinition attribute) throws InternalErrorException {
+		AttributesModuleImplApi module = (AttributesModuleImplApi) getAttributesManagerImpl().getAttributesModule(sess, attribute);
+		if (module == null) {
+			return;
+		}
+
+		Set<AttributeDefinition> moduleDependencies = getDependenciesForModule(sess, module);
+		Set<AttributeDefinition> moduleStrongDependencies = new HashSet<>();
+
+		if (module instanceof VirtualAttributesModuleImplApi) {
+			moduleStrongDependencies = getStrongDependenciesForModule(sess, (VirtualAttributesModuleImplApi) module);
+		}
+
+		dependencies.put(attribute, moduleDependencies);
+		strongDependencies.put(attribute, moduleStrongDependencies);
+
+		generateInverseDependenciesForAttribute(attribute, dependencies, inverseDependencies);
+		generateInverseDependenciesForAttribute(attribute, strongDependencies, inverseStrongDependencies);
+
+		if(isMapOfAttributesDefCyclic(inverseStrongDependencies)) {
+			throw new InternalErrorException("There is a cycle in strong dependencies after adding new attribute definition: " + attribute.getNamespace());
+		}
```

Comment:
This is not correct, because after this exception, database will go for rollback, but data in dependencies stay with this cycle => inconsistent state.

Reply:
Fixed.